examples/nlp/lasertagger/lasertagger_main.py

```diff
@@ -0,0 +1,406 @@
+
+    beam_search = nemo_nlp.nm.trainables.BeamSearchTranslatorNM(
+        decoder=decoder,
+        log_softmax=logits,
+        max_seq_length=args.max_seq_length,
+        beam_size=4,
+        length_penalty=0.0,
+        bos_token=tokenizer.bos_id,
+        pad_token=tokenizer.pad_id,
+        eos_token=tokenizer.eos_id,
+    )
+
+    # tie all embeddings weights
+    if config['use_t2t_decoder']:
+        decoder.tie_weights_with(
+            encoder,
+            weight_names=["embedding_layer.token_embedding.weight"],
+            name2name_and_transform={
+                "embedding_layer.token_embedding.weight": (
+                    "bert.embeddings.word_embeddings.weight",
+                    WeightShareTransform.SAME,
+                )
+            },
+        )
+        decoder.tie_weights_with(
+            encoder,
+            weight_names=["embedding_layer.position_embedding.weight"],
+            name2name_and_transform={
+                "embedding_layer.position_embedding.weight": (
+                    "bert.embeddings.position_embeddings.weight",
+                    WeightShareTransform.SAME,
+                )
+            },
+        )
+
+    def create_pipeline(dataset, num_examples, batch_size, mode):
+
+        data_layer = LaserTaggerDataLayer(
+            dataset, config['use_t2t_decoder'], num_examples, batch_size, mode == "infer"
+        )
+        (input_ids, input_mask, segment_ids, tgt_ids, labels_mask, labels, loss_mask,) = data_layer()
+
+        src_hiddens = encoder(input_ids=input_ids, token_type_ids=segment_ids, attention_mask=input_mask)
+        tgt_hiddens = decoder(
+            input_ids_tgt=tgt_ids, hidden_states_src=src_hiddens, input_mask_src=input_mask, input_mask_tgt=labels_mask
+        )
+        log_softmax = (
+            logits(hidden_states=tgt_hiddens) if config['use_t2t_decoder'] else logits(hidden_states=src_hiddens)
+        )
+        if mode != "infer":
+            loss = loss_fn(logits=log_softmax, labels=labels, loss_mask=loss_mask)
+            per_example_loss = loss_eval_metric(logits=log_softmax, labels=labels, loss_mask=loss_mask)
+        if mode == "infer":
+            if config['use_t2t_decoder']:
+                return [beam_search(hidden_states_src=src_hiddens, input_mask_src=input_mask)]
+            else:
+                return [log_softmax]
+        return loss, [loss, per_example_loss, log_softmax, labels, labels_mask]
+
+    if args.command == "train":
+        # training pipeline
+        train_loss, _ = create_pipeline(train_examples, num_train_examples, args.batch_size, mode="train")
+
+        # evaluation pipelines
+        all_eval_losses = {}
+        all_eval_tensors = {}
+        # for eval_dataset in args.eval_datasets:
+        eval_loss, eval_tensors = create_pipeline(eval_examples, num_eval_examples, args.eval_batch_size, mode="eval")
+        all_eval_losses[0] = eval_loss
+        all_eval_tensors[0] = eval_tensors
```

Comment:
this variable is used in only call: all_eval_tensors[0]

Reply:
yup, removed.